conservator-framework/src/ConservatorFramework.cpp

```diff
@@ -35,9 +43,16 @@ ConservatorFramework::ConservatorFramework(string connectString, int timeout, cl
 }
 
 void ConservatorFramework::start() {
-    //TODO - add some blocking until zoo_state == CONNECTED and throw exception if not connected within timeout
-    this->zk = zookeeper_init(connectString.c_str(), watcher, timeout, cid, 0, 0);
+    unique_lock<std::mutex> connectLock(connectMutex);
+    this->zk = zookeeper_init(connectString.c_str(), watcher, timeout, cid, (void *) this, 0);
+
+    if(!connected) {
+        if(connectCondition.wait_for(connectLock, chrono::seconds(15)) == cv_status::timeout) {
+            throw "timed out waiting to connect to zookeeper";
```

Comment:
Do you need to loop around the check in case of spurious wakeup?

Also, unlock on exception?